### PR TITLE
Use current microstructure in EBSDReaderAvgDataAux

### LINKED ISSUE #7396: Make EBSDReaderAvgDataAux use the current grain structure
### Description of the enhancement or error report
`EBSDReaderAvgDataAux` is not particularly useful right now, as it accesses the _initial_ microstructure from the EBSDReader rather than the EBSD properties mapped to the _current_ microstructure. I'd like to change that.

### Rationale for the enhancement or information for reproducing the error
Make the AuxKernel more useful.

### Identified impact
We'll see. Might need to regold some tests.

Ping @tonkmr, @frombs

### PR DESCRIPTION
**Pull request by dschwen**
While the `EBSDReaderPointDataAux` AuxKernel really only makes sense for accessing the initial EBSD Data, the `EBSDReaderAvgDataAux` AuxKernel should use the current grain structure to look up the averaged per-grain properties. This will allow users to output feature ID maps for evolving mircrostructures.

Closes #7396

### CODE REVIEW

File: modules/phase_field/src/auxkernels/EBSDReaderAvgDataAux.C [hunk 1]
```diff
@@ -5,38 +5,42 @@
...
-  Point p = _current_elem->centroid();
+  // get the dominant grain for the current element/node
+  const int grain_id = _grain_tracker.getEntityValue(isNodal() ? _current_node->id() : _current_elem->id(),
+                                                     FeatureFloodCount::FieldType::UNIQUE_REGION, 0);
 
-  // get the EBSD Point data for the current element
-  const EBSDPointData & d = _ebsd_reader.getData(p);
+  // no grain found
+  if (grain_id < 0)
+    return _invalid;
```
Comment (tonkmr) [inline modules/phase_field/src/auxkernels/EBSDReaderAvgDataAux.C]: Here you are taking the default behavior of getEntityValue in GrainTracker, which returns -1 for areas without active grains, and you make it more general. Should we just change the behavior of getEntityValue to take "invalid" as an argument?

**Comment by tonkmr** [comment]
You have added a new test "1phase_evolution.i" to test EBSDReaderAvgDataAux. I think it might be clearer if the name of the test reflected that. The current test name implies that you are testing the evolution of the reconstructed phase.

**Comment by tonkmr** [comment]
Can you add the feature_id auxvariable to our EBSD reconstruction examples? They are:
modules/phase_field/examples/ebsd_reconstruction/IN100-111grn.i
modules/combined/examples/phase-field_mechanics/EBSD_reconstruction_grain_growth_mech.i

**Comment by permcody** [comment]
I have a PR up that adds it to the former case.

**Comment by dschwen** [comment]
> You have added a new test "1phase_evolution.i" to test EBSDReaderAvgDataAux. I think it might be clearer if the name of the test reflected that. The current test name implies that you are testing the evolution of the reconstructed phase.

The new documentation system links all tests where an object is used automatically. So I really don't want to name specific tests after _one_ of the objects tested in them. The test soes perform a time evolution of the reconstructed microstructure and basically is an extension of the `1phase_reconstruction.i` test.

File: modules/phase_field/src/auxkernels/EBSDReaderAvgDataAux.C (hunk 1 above)
Comment (dschwen) [inline modules/phase_field/src/auxkernels/EBSDReaderAvgDataAux.C]: And what is `getEntityValue` supposed to return? I think this is the right thing to do (and it is basically the same thing you did in your current PR). I'm asking for the current dominant grain and I'm handling the case where no grain is found separately.

Comment (tonkmr) [inline modules/phase_field/src/auxkernels/EBSDReaderAvgDataAux.C]: I'm fine with getEntityValue outputting a -1 when there is no active grain. We can leave it that way.

**Comment by tonkmr** [comment]
👍

**Comment by permcody** [comment]
👍

**Pull request merged**